### PR TITLE
HDDS-9780. Intermittent failure in testS3SecretCacheSizePostDoubleBufferFlush.

### PR DESCRIPTION
**Pull request by ArafatKhan2198**
## What changes were proposed in this pull request?
Since the background thread, `doubleBufferFlush`, flushes the double buffer every **1 second**, when we create a user, it would immediately flush the user to the double buffer, preventing us from verifying the values in the doubleBuffer and cache accurately. Therefore, we have added a trigger to stop it.

Earlier the following error was being thrown by the Test as an Intermittent failure :- 
https://github.com/adoroszlai/ozone-build-results/blob/master/2023/11/27/27054/unit/hadoop-ozone/ozone-manager/org.apache.hadoop.ozone.om.ratis.TestOzoneManagerDoubleBuffer.txt
```
Tests run: 8, Failures: 6, Errors: 0, Skipped: 0, Time elapsed: 10.18 s <<< FAILURE! -- in org.apache.hadoop.ozone.om.ratis.TestOzoneManagerDoubleBuffer
TestOzoneManagerDoubleBuffer.testS3SecretCacheSizePostDoubleBufferFlush -- Time elapsed: 2.087 s <<< FAILURE!
AssertionFailedError: expected: <true> but was: <false>
  ...
  at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:179)
  at org.apache.hadoop.ozone.om.ratis.TestOzoneManagerDoubleBuffer.testS3SecretCacheSizePostDoubleBufferFlush(TestOzoneManagerDoubleBuffer.java:360)
```

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9780

## How was this patch tested?

Tested it over 300 times in the following iterations on my fork :- 
Test Run 1 :- https://github.com/ArafatKhan2198/ozone/actions/runs/7028348940
Test Run 2 :- https://github.com/ArafatKhan2198/ozone/actions/runs/7028351456
Test Run 3 :- https://github.com/ArafatKhan2198/ozone/actions/runs/7028352692

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java [hunk 1]
```diff
@@ -353,7 +357,7 @@ public void testS3SecretCacheSizePostDoubleBufferFlush() throws IOException {
 
     // Flush the current buffer.
     doubleBuffer.flushCurrentBuffer();
-
+    cache = secretManager.cache();
```
Comment (adoroszlai) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java:360]: This part of the change seems to be unnecessary, the object returned by `cache()` is `final`, does not change.

**Review by adoroszlai — COMMENTED** [review]
Thanks @ArafatKhan2198 for the patch.

On a related note, to ensure cleanup is performed even in case of failure, please move these into a `finally` block:

https://github.com/apache/ozone/blob/af0725955031b03898c5ef7dc5f596f88f370e5a/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java#L366-L375

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java (hunk 1 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java]: Done!

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java [hunk 2]
```diff
@@ -340,35 +340,41 @@ public void testS3SecretCacheSizePostDoubleBufferFlush() throws IOException {
...
+      // Flush the current buffer.
+      doubleBuffer.flushCurrentBuffer();
+
+      // Check if all the three secrets are cleared from the cache.
+      Assertions.assertTrue(cache.get(userPrincipalId3) == null);
+      Assertions.assertTrue(cache.get(userPrincipalId2) == null);
+      Assertions.assertTrue(cache.get(userPrincipalId1) == null);
+    } finally {
+      // cleanup metrics
+      doubleBuffer.stopDaemon();
```
Comment (ashishkumar50) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java:368]: Already stopped in first statement of try, so not required to stop again here.

Comment (ArafatKhan2198) [reply hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/ratis/TestOzoneManagerDoubleBuffer.java]: Done!

**Review by ashishkumar50 — COMMENTED** [review]
@ArafatKhan2198 Thanks for updating the patch, LGTM.

**Review by adoroszlai — APPROVED** [review]

**Pull request merged**